src/fs.rs

```diff
@@ -295,6 +295,9 @@ pub trait DavFile: Debug + Send + Sync {
     fn read_bytes(&mut self, count: usize) -> FsFuture<bytes::Bytes>;
     fn seek(&mut self, pos: SeekFrom) -> FsFuture<u64>;
     fn flush(&mut self) -> FsFuture<()>;
+    fn redirect_url(&mut self) -> FsFuture<String> {
```

Comment:
返回 string 再 parse 成 url 还需要处理错误，不如直接要求返回 [`Url`](https://docs.rs/url/2.2.2/url/struct.Url.html)

```suggestion
    fn redirect_url(&mut self) -> FsFuture<Url> {
```